### PR TITLE
Data #50

### PR DESCRIPTION
**Pull request by sdiby**
@ptatian Peter, I've created new vars in the ACS for cost burden, gross rent, and employed persons by major industries. I combined a few of the gross rent levels, but can parse those out if you'd prefer.

### CODE REVIEW

**Comment by ptatian** [comment]
Thank you, @sdiby. I will review.

File: Macros/ACS_summary_geo_source.sas [hunk 1]
```diff
@@ -2657,6 +2716,23 @@
     
     NumRenterHsgUnits_&_years. = NumRenterOccupiedHU_&_years. + NumVacantHsgUnitsForRent_&_years.;
 
+	NumRenterCostBurden_&_years. = sum(B25070e7, B25070e8, B25070e9, B25070e10)
+	NumRentSevereCostBurden_&_years. = B25070e10
+	NumOwnerCostBurden_&_years. = sum(B25091e8, B25091e9, B25091e10, B25091e11, B25091e19, B25091e20, B25091e21, B25091e22)
+	NumOwnSevereCostBurden_&_years. = sum(B25091e11, B25091e22)
+
```
Comment (ptatian) [inline Macros/ACS_summary_geo_source.sas]: We need to add vars for the denominators for these cost burden calculations. The denominator is all renter and all owner units, respectively, excluding the "Not computed" cells.

File: Macros/ACS_summary_geo_source.sas [hunk 2]
```diff
@@ -1947,6 +1962,22 @@
 					sum(C24010Ce7, C24010Ce13, C24010De7, C24010De13, C24010Ee7, C24010Ee13, 
 						C24010Fe7, C24010Fe13, C24010Ge7, C24010Ge13 );
 
+
+		mPopEmployedByInd_&_years. = C24030m1;
+			mPopEmployedAgric_&_years. = %moe_sum( var=C24030m3, C24030m30 );
```
Comment (ptatian) [inline Macros/ACS_summary_geo_source.sas]: There should not be any commas inside the var= parameter. Example:

`mPopEmployedAgric_&_years. = %moe_sum( var=C24030m3 C24030m30 );`

File: Macros/ACS_summary_geo_source.sas [hunk 3]
```diff
@@ -2681,10 +2757,27 @@
     
     mNumRenterHsgUnits_&_years. = %moe_sum( var=mNumRenterOccupiedHU_&_years. mNumVacantHUForRent_&_years. );
 	
+	mNumRenterCostBurden_&_years. = %moe_sum( var=B25070m7, B25070m8, B25070m9, B25070m10)
```
Comment (ptatian) [inline Macros/ACS_summary_geo_source.sas]: Similar comment as above. No commas after var=.

File: Macros/Acs_sf.sas [hunk 4]
```diff
@@ -93,6 +93,7 @@
   03/04/14 MSW Added sequence numbers 0063, 0064 and 0075. Added  B01002Ee/m, B25014Ee/m, and B19013Ee/m to drop list. 
   11/05/14 SXZ Added Table B19113 B09001
   06826/17 LP and RP Added Tables for racial equity work.
+  07/26/17 Somala Diby added B25106 for Bridge Park tabulations.
```
Comment (ptatian) [inline Macros/Acs_sf.sas]: I guess you didn't actually have to modify this program.

**Comment by sdiby** [comment]
Thanks @ptatian I've committed the above changes to this branch under "Responding to @ptatian 's changes," ready for your review.

**Comment by ptatian** [comment]
@sdiby There are still a few syntax errors. 

```
MPRINT(ACS_SUMMARY_GEO_SOURCE):   mPopEmployedEduction_2011_15
NOTE 138-205: Line generated by the macro variable "_YEARS".
89             mPopEmployedEduction_2011_15
               ____________________________
               22
ERROR 22-322: Syntax error, expecting one of the following: !, !!, &, *, **, +, -, /, ;, <, <=, <>, =, >, ><, >=, AND, EQ, 
              GE, GT, IN, LE, LT, MAX, MIN, NE, NG, NL, NOTIN, OR, ^=, |, ||, ~=.  
```
You can test the program by submitting D:\DCData\Libraries\ACS\Prog\SF_2011_15\ACS_2011_15_dc_sum_all.sas. It's OK to submit the code as long as you are doing an **interactive SAS session** or a **batch submit from D:** because no files will be replaced on L:.

**Comment by ptatian** [comment]
Sorry, @sdiby. I forgot to note earlier that it appears that the Bridge Park summary is not included in the list of summary files being created. In D:\DCData\Libraries\ACS\Macros\ACS_summary_geo_source.sas, you need to add the following at line 2889:

`    %ACS_summary_geo( bpk, &source_geo )`

**Comment by sdiby** [comment]
Thanks @ptatian  I've added Bridge Park line to the list of summary files via commit.

**Comment by ptatian** [comment]
Thanks, @sdiby, but I'm still getting errors. You have a number of statements with missing semicolons in Acs_summary_source_geo, for example: 

`	OwnerCostBurdenDenom_&_years. = NumOwnerOccupiedHU_&_years. - sum(B25091e12, B25091e23)`

Also, just realized another thing. Since we will be updating data sets that were created previously, we should include an appropriate note in the metadata. For example, in ACS_2011_15_dc_sum_all.sas, we should add a revisions= line to the %ACS_summary_all() macro invocation:

```
%ACS_summary_all( 

  /** State abbreviation. Ex: DC **/
  state_ab = DC,

  /** Year range (xxxx_yy). Ex: 2005_09 **/
  years = 2011_15,

  revisions=Add workers by industry/occupation, renters by gross rent, and housing cost burdens.

)
```

**Please note the comma (,) after years=2011_15.**

**Comment by ptatian** [comment]
@sdiby Please add the revisions= line to **ACS_2011_15_va_sum_all.sas** and **ACS_2011_15_md_sum_all.sas** as well. Thanks!

**Comment by ptatian** [comment]
Sigh...I should have written:

```
%ACS_summary_all( 

  /** State abbreviation. Ex: DC **/
  state_ab = DC,

  /** Year range (xxxx_yy). Ex: 2005_09 **/
  years = 2011_15,

  revisions=%str(Add workers by industry/occupation, renters by gross rent, and housing cost burdens.)

)
```

**Comment by sdiby** [comment]
@ptatian @rpitingolo 
Rob, these changes are ready for your review:
- added missing semicolons in summary_geo_source macro
- added missing commas to sum_all progs
- added revisions line to sum_all_progs

**Comment by rpitingolo** [comment]
@sdiby I'll review.

**Comment by rpitingolo** [comment]
@sdiby I fixed some typos that were causing errors. I'm getting a " Variable B2506314 is uninitialized." note. It looks like you need these cells for some of your new variables. Was it added correctly in the SF programs?

**Comment by sdiby** [comment]
@rpitingolo I've fixed the variable--it was read "B2506314" and should've read "B25063e14". I also fixed the corresponding MOE var name.

**Comment by rpitingolo** [comment]
@sdiby OK. Glad it wasn't too complicated. Please go ahead and merge/finalize.

**Pull request merged**